src/sources/aws_s3/sqs.rs

```diff
@@ -182,7 +183,8 @@ impl Ingestor {
         for _ in 0..self.state.client_concurrency {
             let process =
                 IngestorProcess::new(Arc::clone(&self.state), out.clone(), shutdown.clone());
-            let handle = tokio::spawn(async move { process.run().await });
+            let fut = async move { process.run().await };
+            let handle = tokio::spawn(fut.in_current_span());
```

Comment:
I'm wondering if we are missing this sort of thing in other places too.

Reply:
Yeah, it's a bit insidious.

Reply:
Oh dear. Absent some means to automatically catch this I'd stump for an audit ticket.

Reply:
👍 opened https://github.com/timberio/vector/issues/7970